src/relay.rs

```diff
@@ -77,6 +80,9 @@ impl<T: DuplexTransport + 'static> Relay<T> {
             .join(self.sidechain.handle_missed_transfers(&self.homechain, handle))
             .join(self.handle_requests(rx, handle))
             .and_then(|_| Ok(()))
+            .map_err(|_| {
+                process::exit(1);
```

Comment:
should exit with -1 if there's an issue
```suggestion
                process::exit(-1);
```